src/utilities/geometry/Polygon3d.hpp

```diff
@@ -0,0 +1,117 @@
+*  may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without specific prior
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#ifndef UTILITIES_GEOMETRY_POLYGON_HPP
+#define UTILITIES_GEOMETRY_POLYGON_HPP
+
+#include "../UtilitiesAPI.hpp"
+
+#include "Point3d.hpp"
+#include <vector>
+
+namespace openstudio {
+
+class Vector3d;
+
+class UTILITIES_API Polygon3d
+{
+ public:
+  // Constructs an empty polygon
+  Polygon3d();
+
+  // Construct a polygon with an outer path
+  Polygon3d(const Point3dVector& outerPath);
+
+  // Constructs a polygon with an outer path and one or more inner paths
+  Polygon3d(const Point3dVector& outerPath, const Point3dVectorVector& innerPaths);
+
+  // Assigns an outer path for the polygon
+  void setOuterPath(const Point3dVector& outerPath);
+
+  // Returns the polygon's outer path
+  Point3dVector getOuterPath() const;
+
+  // Returns the polygon's inner paths
+  Point3dVectorVector getInnerPaths() const;
+
+  // Adds a point to the outer path of the polygon
+  void addPoint(const Point3d& point);
+
+  // Adds an inner path to the polygon
+  void addHole(const Point3dVector& hole);
+
+  // Calculates the Newall Vector for the polygon
+  Vector3d newallVector();
+
+  // Calculates the outward normal of the polygon
+  Vector3d outwardNormal();
+
+  // Calculates the gross area of the polygon (area excluding holes)
+  double grossArea();
+
+  // Calculates the net area of the polygon (gross area - area of holes)
+  double netArea();
+
+  // Gets the perimeter of the outer path of the polygon
+  double perimeter();
+
+  // Determines whether the polygon os clockwise (normal down) or anti-clockwise (normal up)
+  bool isClockwise();
+
+  // Calculates the centroid of the polygon
+  Point3d centroid();
+
+  // Point is on the outer path or one of the inner paths
```

Comment:
Shouldn't the inner paths be inside the outer path?  Do you mean "on the outer path" or "inside the outer path"?

Reply:
Depends on if you need define inside being within the polygon bounds but not on one of the paths or whether inside is defined as being within the polygon bounds or on one of the paths. The calling context would dictate that so these methods just provide the flexibility

And also you may want to know only whether a point is on one of the paths